nav2_behaviors/src/behavior_server.cpp

```diff
@@ -111,7 +136,12 @@ BehaviorServer::loadBehaviorPlugins()
         get_logger(), "Creating behavior plugin %s of type %s",
         behavior_ids_[i].c_str(), behavior_types_[i].c_str());
       behaviors_.push_back(plugin_loader_.createUniqueInstance(behavior_types_[i]));
-      behaviors_.back()->configure(node, behavior_ids_[i], tf_, collision_checker_);
+      behaviors_.back()->configure(
```

Comment:
I'd like a new Behavior plugin API to return an enum 

```
enum class CostmapInfoType {
  NONE = 0,
  LOCAL = 1,
  GLOBAL = 2,
  BOTH = 3
};

...

CostmapInfoType getRequiredCostmapInfoTypes() {return CostmapInfoType::LOCAL;}
```

From that, we can query all the plugins as to which they need. If there are any (global or local) that are not required by any plugins, then that collision checker, footprint sub, and costmap sub can be reset. Especially the costmap sub for the global costmap is non-trivial in its overhead if no one requires it.

Reply:
You don't need to use those exact class/function names, I was just trying to be illustrative

Reply:
Alright, 

That sounds good.